### PR TITLE
[BE]Task #213077  Acknowledgement create api

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/213077

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **New Features**
    - Integrated `moment-timezone` for handling timezones, specifically setting to Indian Standard Time for acknowledgment records.
- **Chores**
    - Updated dependencies, including `axios` and `follow-redirects`, for improved performance and security.

### CODE REVIEW

File: src/src/userauth/userauth.service.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { UserHelperService } from 'src/helper/userHelper.service';
 import { HasuraService } from 'src/services/hasura/hasura.service';
 import { KeycloakService } from 'src/services/keycloak/keycloak.service';
 import { AuthService } from 'src/modules/auth/auth.service';
+const moment = require('moment-timezone');
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/userauth/userauth.service.ts:7]: fix this

File: src/src/userauth/userauth.service.ts [hunk 2]
```diff
@@ -154,6 +155,38 @@ export class UserauthService {
...
+					//add acknowledgment details
+					await this.hasuraService.q(
+						'acknowledgement',
+						{
+							...acknowledgement_create_body,
+						},
+						[],
+						false,
+						['id'],
+					);
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/userauth/userauth.service.ts:187]: @GitVivekHub create core service for this and call and use it from other services.

File: src/src/common/method/method.ts [hunk 3]
```diff
@@ -115,4 +116,16 @@ export class Method {
...
+		return new Promise((resolve, reject) => {
+			try {
+				const istTime = moment().tz('Asia/Kolkata');
+				const formattedISTTime = istTime.format('YYYY-MM-DD HH:mm:ss');
+				resolve(formattedISTTime);
+			} catch (error) {
+				reject(error);
+			}
+		});
+	}
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/common/method/method.ts:130]: fix this @GitVivekHub

File: src/src/userauth/userauth.service.ts [hunk 4]
```diff
@@ -154,6 +156,36 @@ export class UserauthService {
...
+				let user_id = result?.data?.id;
+
+				if (user_id) {
+					// Set the timezone to Indian Standard Time (Asia/Kolkata)
+					const formattedISTTime =
+						await this.method.getFormattedISTTime();
+
+					// Format the time as per datetime
+
+					let acknowledgement_create_body = {
```
Comment (manojLondhe) [inline src/src/userauth/userauth.service.ts:168]: @GitVivekHub create core service for this and call and use it from other services.

File: src/src/modules/acknowledgement/acknowledgement.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+				[],
+				false,
+				['id'],
+			);
+			return result;
+		} catch (error) {
+			return error;
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/modules/acknowledgement/acknowledgement.service.ts]: no need

**Pull request merged**